chain/rust/src/byron/utils.rs

```diff
@@ -153,9 +153,9 @@ impl AddressContent {
     }
 
     // icarus-style address (Ae2)
```

Comment:
Note: I started this PR thinking that those 6 simultaneous PRs had a compile error after merging them together and this would be to fix that, but after realized this fix is in #227 and it just wasn't merged yet. It's the same change here/there so it should be fine though. Just had to change the param type to fix the WASM bindings. The rest of the PR is just fixing warnings though.